### ISSUE TITLE
Disable self notification on requireInputText

### ISSUE THREAD

**Issue by abhay-s-rawat** [rafaelsetragni/awesome_notifications#563]
Hi I am using reply button as below but the problem is a self notification gets generated on reply which does not trigger onNotificationCreatedMethod, onNotificationDisplayedMethod.
Can I disable that self notification and sometimes the progress indicator keeps rotating but that self notification does not gets generated.
```
NotificationActionButton(
          key: chatReplyButtonKey,
          label: 'Reply',
          color: Colors.blue,
          autoDismissible: false,
          showInCompactView: true,
          requireInputText: true,
        ),
```
![Screenshot_1664256295](https://user-images.githubusercontent.com/84996356/192439341-ee1e1be2-82be-48dc-9011-3eb4482b4456.png)

**Comment by rafaelsetragni** [comment]
The self notification is necessary to stop the spinning indicator. Is required to update the notification, but i don't know how the id inside the reply are getting different in the second notification. Can you provide me the full example of how you create the notification channel and this notification?

Also, i created a template to guide you on how to report your issue. If you prefer, recreate this issue using the new template.

**Comment by abhay-s-rawat** [comment]
On rejectandreply button I am getting a text input from user and I remove the notification but suddenly another notification gets generated which is exactly same.
So when hitting send button if I add a 3 seconds delay then its ok but if I remove delay the notification which I removed gets generated automatically.
```
await Future.delayed(const Duration(seconds: 2));
await notif.removeCallNotification(action.payload!["callId"]!);
```
```
Future<bool> callNotification({
    bool videoCall = false, // data session and ongiong call notification
    required String loggedInUser,
    required String from,
    required String fromName,
    required String callId,
    String? picUrl,
  }) async {
    int id = DateTime.now().toUtc().millisecondsSinceEpoch ~/ 1000;
    bool success = await AwesomeNotifications().createNotification(
      content: NotificationContent(
        id: id,
        channelKey: AwesomeNotificationsHelper.instance.callChannelKey,
        color: Colors.blue,
        summary: "Incoming call",
        //backgroundColor: Colors.green,
        //ticker: "ticker string",
        body: '$fromName ($from) : $loggedInUser',
        title:
            '${videoCall ? Emojis.video_movie_camera : Emojis.phone_telephone_receiver} Incoming Call...',
        actionType: ActionType.Default,
        //groupKey: AwesomeNotificationsHelper.instance.callGroupKey,
        wakeUpScreen: true,
        category: NotificationCategory.Call,
        locked: false,
        autoDismissible: false,
        //fullScreenIntent: true,
        payload: {
          "loggedInUser": loggedInUser,
          "from": from,
          "callId": callId,
          "picUrl": picUrl,
          "fromName": fromName,
          "videoCall": videoCall ? "true" : "false",
        },
        largeIcon: picUrl,
      ),
      actionButtons: [
        NotificationActionButton(
          key: callAnswerButtonKey,
          label: "Answer",
          color: Colors.green,
          autoDismissible: false,
          showInCompactView: true,
        ),
        NotificationActionButton(
          key: callIgnoreButtonKey,
          label: "Dismiss",
          color: Colors.grey,
          autoDismissible: false,
          showInCompactView: true,
        ),
        NotificationActionButton(
          key: callRejectButtonKey,
          label: "Reject",
          color: Colors.red,
          autoDismissible: false,
          showInCompactView: true,
        ),
        NotificationActionButton(
          key: callRejectWithReplyButtonKey,
          label: "Reject and reply",
          requireInputText: true,
          color: Colors.grey,
          autoDismissible: false,
          showInCompactView: true,
          actionType: ActionType.KeepOnTop,
        ),
      ],
    );
    if (success) {
      log("call map created");
      callNotificationsMap[callId] = id;
    }
    return success;
  }
```

**Comment by abhay-s-rawat** [comment]
Which branch has latest code ? I cannot see the implementation of requireInputText.
awesome_notifications: ^0.7.0-beta.7+5

**Comment by abhay-s-rawat** [comment]
By default self notification title is summary of notification can we have different property for it too ? Like "me" word will look great while quick reply to chats.

**Comment by abhay-s-rawat** [comment]
Sorry for not knowing what exactly is happening in instant reply button, I went through following video and came to tknow that the duplicate notifications were coming up in such case was due to the updation of the notification. So I guess waiting for notification to get updated will be better like I did with future.delayed. 
Still 1 doubt is there , when notification gets updated why other callback are not getting fired, Is there a update event which can be tracked ?
https://www.youtube.com/watch?v=DsFYPTnCbs8

A minor enhancement request from my side will be to add a different title for self notification by default it is taking summary as title in grouped notifications. Can this be done please ?

**Comment by abhay-s-rawat** [comment]
By default on reply a circular progress indicator starts spinning so on action button press I am removing the notification which creates this self notification otherwise it keeps spinning forever, is that desired functionality ?

**Comment by rafaelsetragni** [comment]
> By default self notification title is summary of notification can we have different property for it too ? Like "me" word will look great while quick reply to chats.

The last code is ^0.7.0-beta.7+5 and you can find a complete example of each feature in AwN in our example project.
In the example app there is an example of chat messages with reply implemented, and there i used "Me" to represent your own message.

> Which branch has latest code ? I cannot see the implementation of requireInputText.
> awesome_notifications: ^0.7.0-beta.7+5

Is this one. And also there is requireInputText. The deprecation happened for the layout `InputText`, because now you can combine requireInputText property with all other layouts.

**Comment by abhay-s-rawat** [comment]
I cannot find branch for 0.7.0-beta.7+5. I went through examples after that I wrote above code, the were very good .
Can I pass some parameters when notification gets tapped and app is launched. I am using fcm to handles backgroud messages there I am generating custom notifications, If I can pass parameter I can use it to route to different pages. I am using auto route.
Thanks a lot for the time.
